test/run-unittest.d

```diff
@@ -0,0 +1,111 @@
+#!/usr/bin/env dub
+/+dub.sdl:
+	name: run_unittest
+	targetName: run-unittest
++/
+module run_unittest;
+
+/// Name of the log file
+enum logFile = "test.log";
+
+/// has true if some test fails
+bool any_errors;
+
+/// prints (non error) message to standard output and log file
+void log(Args...)(Args args)
+	if (Args.length)
+{
+	import std.format : format;
+	import std.stdio : File, stdout, writeln;
+
+	version(Windows) const str = format("[INFO] " ~ args[0], args[1..$]);
+	else const str = format("\033[0;32m[INFO] " ~ args[0] ~ "\033[0m", args[1..$]).format(args[1..$]);
+	writeln(str);
+	stdout.flush;
+	File(logFile, "a").writeln(str);
+}
+
+/// prints error message to standard error stream and log file
+/// and set any_errors var to true value to indicate that some
+/// test fails
+void logError(Args...)(Args args)
+{
+	import std.format : format;
+	import std.stdio : File, stderr;
+
+	version(Windows) const str = format("[ERROR] " ~ args[0], args[1..$]);
+	else const str = format("\033[0;31m[ERROR] " ~ args[0] ~ "\033[0m", args[1..$]).format(args[1..$]);
+	stderr.writeln(str);
+	stderr.flush;
+	File(logFile, "a").writeln(str);
+	any_errors = true;
+}
+
+int main(string[] args)
+{
+	import std.algorithm : among;
+	import std.file : dirEntries, DirEntry, exists, getcwd, readText, SpanMode;
+	import std.format : format;
+	import std.stdio : File, writeln;
+	import std.path : absolutePath, buildNormalizedPath, baseName, dirName;
+	import std.process : environment, spawnProcess, wait;
+
+	//** if [ -z ${DUB:-} ]; then
+	//**     die $LINENO 'Variable $DUB must be defined to run the tests.'
+	//** fi
+	auto dub = environment.get("DUB", "");
+	writeln("DUB: ", dub);
+	if (dub == "")
+	{
+		logError("Environment variable `DUB` must be defined to run the tests.");
```

Comment:
Mixed up the quotes. Do you mean
```d
logError(`Environment variable "DUB" must be defined to run the tests.`);
```
?

Reply:
Fixed

Reply:
I just learned that using backticks inside strings makes sense when the compiler produces the output, ref https://github.com/dlang/dub/pull/2095#discussion_r563411841. However here the output is produced by the script, and so it would not do any syntax highlighting. So I still think using backticks as the outer level quotes is what is wanted here, to produce a [WYSIWYG string](https://dlang.org/spec/lex.html#wysiwyg).